src/image-upload/image.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { Headers, Http, RequestOptions, RequestOptionsArgs, Response } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
+
+@Injectable()
+export class ImageService {
+
+  constructor(private http: Http) { }
+
+  public postImage(url: string, image: File, headers?: any[], partName: string = 'image',
+    withCredentials?: boolean): Observable<Response> {
+
+    if (!url || url === '') {
+      throw new Error('Url is not set! Please set it before doing queries');
+    }
+
+    let options: RequestOptionsArgs = new RequestOptions();
+    if (withCredentials) {
+      options.withCredentials = withCredentials;
+    }
+
+    if (headers) {
+      options.headers = new Headers();
+      for (let header of headers) {
+        let key = Object.keys(header)[0];
+        options.headers.append(key, header[key]);
+      }
+    }
```

Comment:
Okay I've been poking around in angular - it expects either a `Headers` object or a dictionary, which is why it wasn't happy with the array.

I suggest we just use this signature wherever we need headers, and get rid of the interface (sorry, I know I asked you to do it):

`Headers|{[name: string]: any}`

Then the following will work:

````
myFunc(headers: Headers|{[name: string]: any}) {
  const options: any = {};
  options.headers = new Headers(headers);
}
````

The user can pass this in by instantiating their own `Headers` object or something like:

````
const headers: {[name: string]: any} = {
  potato: 'cheese',
  cheese: 'potato'
};
````